ecl/hql/hqlgram.y

```diff
@@ -8812,6 +8812,10 @@ simpleDataSet
 
                             bool hasFileposition = getBoolAttributeInList(extra, filepositionAtom, true);
                             record.setown(parser->checkIndexRecord(record, $5, extra));
+
+                            if (!parser->checkAllowed($1, "datafile", "Datafile access"))
+                                extra.setown(createComma(extra.getClear(), createAttribute(_disallowed_Atom)));
```

Comment:
Yes I realize that.
I meant that it's doing the checking process, the checkAllowed() / allowAccess(), that iterates deniedPermissions, allowSignedPermissions and allowedPermissions arrays here at parse time, in order to flag..

Could it delay and check later if/when actually used just before firing the error?